lib/pkcs7.js

```diff
@@ -537,6 +548,134 @@ p7.createSignedData = function() {
     // add signer info
     msg.signerInfos = _signersToAsn1(msg.signers);
   }
+
+  function verifySignerInfos(mds, caStore, options) {
```

Comment:
I see you're providing for `options.callback`.  This seems to me to be an unconventional way of implementing a callback which would not be compatible with `util.promisify` implementations which expect the last param to be a callback function.

That said, can you say more about the use case for callbacks here?  Given that all the code here is synchronous, does the callback add value?

If the callback is going to remain part of this API, it appears to me that it is not being called in a number of situations.  For example, if the conditional on L556 fails we end up at L677 `retrurn rval;` and the callback is never called.  Also, for the most part this function is written to throw errors at various times when it may be appropriate to call the callback with `callback(err);` instead.

Do you think it would be simpler to eliminate the callback functionality?

Reply:
I added a callback so that as a part of the verification process you would get information on good signatures to display to the user.  I would like to keep it for that reason.  I'll clean it up and make it promisifyable and we'll see how it looks.

Reply:
The purpose of the callback isn't to get a result so much as status updates on each signature processed.  The verify function could return an array of results, yes.  At that point I would just eliminate the callback.  Is that what you would prefer?

Reply:
I understand now that my confusion has been caused by referring to this passed in function as a `callback`.  I think of a `callback` as a function that is called when the called function completes/errors.  This is reflected in the `util.promisify` API such that the Promise is resolved when the callback is invoked, which is *not* what you are attempting to do here.

If this were a long running asynchronous process, a solution involving event emitters might be appropriate.  

Now that I understand what you're trying to do and `options.onSignatureVerificationComplete` handler might be appropriate.

Do you have some use case that requires there to be status updates on each signature?  If not, then I think removing the status callback function would be best.

Reply:
If I modify the return to contain the same info as the status callback/emitter then no, I don't need it.  I'll do that now.

Reply:
I ended up going with the per-signature event emitter vs a return of an array of results.  Let me know what you think.

Reply:
There should be a common pattern in the whole lib for events and more detailed results.  That's a bigger project.  This does do the common return true/false and that will work for now.